.github/workflows/main.yml

```diff
@@ -66,8 +68,35 @@ jobs:
            rm -rf CMakeFiles
            find . -maxdepth 1 -type f -not -name 'contrib_build.log' -delete
 
-    #TODO switch to V2 once it is released (this is to only upload relevant folders)
-    - uses: actions/upload-artifact@v3
+    - name: Configure Git for Tagging
+      if: github.event_name == 'workflow_dispatch'
+      run: |
+        git config --global user.name 'GitHub Actions'
+        git config --global user.email 'actions@github.com'
+
+    - name: Create and Push Tag
+      if: github.event_name == 'workflow_dispatch'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        TAG_NAME="release-$(date +'%Y%m%d%H%M%S')"
```

Comment:
not sure how to create a good tag name here or how to properly name files:
Currently I use:
./contrib-build/contrib_build-${{runner.os}}.tar.gz
for the zip.
Note that we need a way to download the proper one from the github actions in OpenMS/OpenMS

Reply:
I don't see the naming here as especially important, we generate the tags that we use publicly on running the release action, so these are just for internal use right? The way I see it we can just have the OpenMS/OpenMS action grab the latest tag, and use the contrib file that corresponds to its own OS. Or am I missing something?

Reply:
yeah maybe it is enough to just distinguish by os (and hope that the runner are all the same).
On jenkins we observed a lot of minor changes between XCode and VSCode versions that required individual contrib packages (each one was in a different folder on abibuilder)
We might just try it and then get the release with something like:
```
name: Fetch Latest Release

on:
  workflow_dispatch:

jobs:
  fetch-release:
    runs-on: ubuntu-latest
    steps:
      - name: Fetch latest release from another repository
        id: get-latest-release
        uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.GH_TOKEN }}
          script: |
            const repoOwner = 'other-repo-owner';  // Replace with the owner of the repository
            const repoName = 'other-repo-name';    // Replace with the repository name
            const response = await github.repos.getLatestRelease({
              owner: repoOwner,
              repo: repoName
            });
            return response.data;

      - name: Use the release data
        run: |
          echo "The latest release tag is ${{ steps.get-latest-release.outputs.tag_name }}"
          echo "Download URL is ${{ steps.get-latest-release.outputs.zipball_url }}"
```
and then just extract / use the one matching the OS?
No idea if this works :-P